### PR TITLE
Private events APP development

### PR DESCRIPTION
**Pull request by carlos-osorio-developer**
## Add features for:

- User MVC.
- Event MVC.
- Associations between both models with has_many through the "Attending" table.
- Authentication methods.
- Navbar and navigation.
- Tests.

### CODE REVIEW

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUESTED🛑🛑🛑**

Hello @carlos-osorio-developer 
Great work making this project.
I can tell you put in a lot of effort. I like how events are separated.
However, let's make minor changes and get it approved.

[REQUIRED]: The attend button is not working. Kindly fix it.
[REQUIRED]: The logout button is not working. Kindly fix it.
[REQUIRED]: Kindly add unit and integration tests.

You can find the other requirements below.

Make the required changes and ask for another review👍👍
Happy coding💻💻

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user!, only: [:new]
+  def index
+    @events = Event.all
+  end
```
Comment (NtwaliHeritier) [inline app/controllers/events_controller.rb:5]: The index action should have these two lines. 
lists past events, i.e. @past_events = Event.previous_events (screenshot from Odin)
lists ucomping events, i.e. @upcoming_events = Event.upcoming_events (screenshot from Odin)

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by_id(params[:id])
+  end
```
Comment (NtwaliHeritier) [inline app/controllers/users_controller.rb:8]: The show action should have these lines.

lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

Comment (carlos-osorio-developer) [reply app/controllers/users_controller.rb:8]: > The show action should have these lines.
> 
> lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
> 
> lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
> 
> lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

 - Can you be more specific with "logout button is not workin"? It's working here.
 - What do you mean by "unit and integration test"?
- Also, I can not see any of those screenshots in the project specifics: https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/associations

Thanks.

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUESTED🛑🛑**

Hello again @carlos-osorio-developer 
Great work making most of the required changes.
However to get it approved you must make all the required changes👍👍

Kindly make all the changes required in the previous review and ask for another review👌👌

Happy coding💻💻

**Comment by carlos-osorio-developer** [comment]
Sorry but those changes don't reflect the specifications provided by the documentation, please go to the link and check it yourself.

https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/associations

**Comment by carlos-osorio-developer** [comment]
Sorry but those changes don't reflect the specifications provided by the documentation, please go to the link and check it yourself.

https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/associations

The changes that you are asking for are not in the project specifications.

**Comment by carlos-osorio-developer** [comment]
image

image

**Comment by carlos-osorio-developer** [comment]
Kindly look at the screenshots from the original website.

Kindly enter the link provided before. 

https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/associations

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (sinansevgi) [reply app/controllers/events_controller.rb:5]: - [ ] Please check the project specification link that I share. You can see this requirement in there

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (sinansevgi) [reply app/controllers/users_controller.rb:8]: - [ ] Please check the project specification link that I share. You can see these requirements there

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @carlos-osorio-developer , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

### Looks like The Odin Project updated their project specification 2 days ago. Please check [this link ](https://github.com/TheOdinProject/curriculum/blob/e71e03d984bc0b2c3d251a7cbcf9721c796b4175/rails_programming/advanced_forms_and_activerecord/project_associations.md) for previous description. 

I am so sorry for this confusion. 

## Required Changes ♻️
_Please check the inline comments made by previous TSE_ 

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (sinansevgi) [reply app/controllers/users_controller.rb:8]: - [x] Please check the project specification link that I share. You can see these requirements there

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (sinansevgi) [reply app/controllers/events_controller.rb:5]: - [ ] Please check the project specification link that I share. You can see these requirements there

**Comment by carlos-osorio-developer** [comment]
If The Odin Project updated their project, you should update your reviews. I had no way to know that before doing the project. Use your common sense.

Also, the requirements of the previous app and the current one are the same, and they function exactly the same.

**Review by emmanuellekamwa — APPROVED** [review]
Hi team. Congrats on implementing all the necessary changes 👏🏽👏🏽. Your project looks great and it is approved 🎉🎉. Good luck with the next projects and happy coding 👨🏽‍💻👨🏽‍💻🚀🚀

**Pull request merged**